app/workers/builds.php

```diff
@@ -0,0 +1,262 @@
+        \curl_setopt($ch, CURLOPT_POST, true);
+        \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        \curl_setopt($ch, CURLOPT_TIMEOUT, 900);
+        \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+        \curl_setopt($ch, CURLOPT_HTTPHEADER, [
+            'Content-Type: application/json',
+            'x-appwrite-project: '.$projectId,
+            'x-appwrite-executor-key: '. App::getEnv('_APP_EXECUTOR_SECRET', '')
+        ]);
+        \curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode([
+            'functionId' => $functionId,
+            'tagId' => $tagId
+        ]));
+
+        $response = \curl_exec($ch);
+        $responseStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        $error = \curl_error($ch);
+        if (!empty($error)) {
+            throw new \Exception($error);
+        }
+
+        \curl_close($ch);
+
+        if ($responseStatus !== 200) {
+            throw new \Exception("Build failed with status code: $responseStatus");
+        }
+
+        $response = json_decode($response, true);
+        if (isset($response['error'])) {
+            throw new \Exception($response['error']);
+        }
+
+        if (isset($response['success']) && $response['success'] === true) {
+            return;
+        } else {
+            throw new \Exception("Build failed");
+        }
+    }
+
+    protected function buildTag(string $projectId, string $functionId, string $tagId) 
+    {
+        $dbForProject = $this->getProjectDB($projectId);
+        
+        // TODO: Why does it need to skip authorization?
+        $function = Authorization::skip(fn() => $dbForProject->getDocument('functions', $functionId));
+
+        if ($function->isEmpty()) {
+            throw new Exception('Function not found', 404);
+        }
+
+        // Get tag document
+        $tag = $dbForProject->getDocument('tags', $tagId);
+        if ($tag->isEmpty()) {
+            throw new Exception('Tag not found', 404);
+        }
+
+        $runtimes = Config::getParam('runtimes', []);
+        $key = $function->getAttribute('runtime');
+        $runtime = isset($runtimes[$key]) ? $runtimes[$key] : null;
+        if (\is_null($runtime)) {
+            throw new Exception('Runtime "' . $function->getAttribute('runtime', '') . '" is not supported');
+        }
+
+        $buildId = $tag->getAttribute('buildId', '');
+
+        // If build ID is empty, create a new build
+        if (empty($buildId)) {
+            try {
+                $buildId = $dbForProject->getId();
+                // TODO : There is no way to associate a build with a tag. So we need to add a tagId attribute to the build document
+                // TODO : What should be the read and write permissions for a build ? 
+                $dbForProject->createDocument('builds', new Document([
+                    '$id' => $buildId,
+                    '$read' => ['role:all'],
+                    '$write' => ['role:all'],
```

Comment:
What should the read and write permissions be for a build ?

Reply:
After a discussion, it was decided to have these permissions empty. Which means we the builds documents cannot be read through an API